File: jats-demo-webapp/src/main/ts/model/Component.ts
class Component
{
	private price : number;

	private requiredComponents : Component[];

	private optionalComponents : Component[];

	private modifiers : Modifier[];

	private id : number;

	private label : string;

	private quantity : Quantity;

	public getPrice() { return this.price; }
	public setPrice( price : number ) { this.price = price; }

	public getRequiredComponents() { return this.requiredComponents; }
	public setRequiredComponents( requiredComponents : Component[] ) { this.requiredComponents = requiredComponents; }

	public getOptionalComponents() { return this.optionalComponents; }
	public setOptionalComponents( optionalComponents : Component[] ) { this.optionalComponents = optionalComponents; }

	public getModifiers() { return this.modifiers; }
	public setModifiers( modifiers : Modifier[] ) { this.modifiers = modifiers; }

	public getId() { return this.id; }
	public setId( id : number ) { this.id = id; }

	public getLabel() { return this.label; }
	public setLabel( label : string ) { this.label = label; }

	public getQuantity() { return this.quantity; }
	public setQuantity( quantity : Quantity ) { this.quantity = quantity; }


}

File: jats-demo-model/src/main/java/com/github/pepe79/jats/demo/model/Component.java
package com.github.pepe79.jats.demo.model;

import java.util.List;

public class Component
{
	private int id;

	private String label;

	private float price;

	private List<Component> requiredComponents;

	private List<Component> optionalComponents;

	private List<Modifier> modifiers;

	private Quantity quantity;

	public float getPrice()
	{
		return price;
	}

	public void setPrice(float price)
	{
		this.price = price;
	}

	public List<Component> getRequiredComponents()
	{
		return requiredComponents;
	}

	public void setRequiredComponents(List<Component> requiredComponents)
	{
		this.requiredComponents = requiredComponents;
	}

	public List<Component> getOptionalComponents()
	{
		return optionalComponents;
	}

	public void setOptionalComponents(List<Component> optionalComponents)
	{
		this.optionalComponents = optionalComponents;
	}

	public List<Modifier> getModifiers()
	{
		return modifiers;
	}

	public void setModifiers(List<Modifier> modifiers)
	{
		this.modifiers = modifiers;
	}

	public int getId()
	{
		return id;
	}

	public void setId(int id)
	{
		this.id = id;
	}

	public String getLabel()
	{
		return label;
	}

	public void setLabel(String label)
	{
		this.label = label;
	}

	public Quantity getQuantity()
	{
		return quantity;
	}

	public void setQuantity(Quantity quantity)
	{
		this.quantity = quantity;
	}

}
